### ISSUE TITLE
ethnode: New geth uses new ID that is not the enode pubkey

### ISSUE THREAD

**Issue by shazow** [vipnode/vipnode#58]
Our code assumes that admin.peers[].id is the enode ID, and comparisons break.

**Comment by mining-visualizer** [comment]
I was just trying vipnode now with the latest version of geth.  I think maybe I'm being affected by this issue.  At least it doesn't seem to be working to me.  Vipnode is just showing a bunch of lines like this:
```
[agent] 2019/07/11 16:18:19 Pool update: peers=0 active=0 invalid=0 block=0 balance=Balance(<null account>, 0 wei
```

And geth is not syncing at all.  Do these symptoms match your bug?  Any idea when a fix will be available?

**Comment by mining-visualizer** [comment]
Btw I also noticed that the latest version of vipnode does not have those nice instructions about how to get to the web site to add some funds.  Maybe put that back in,  or put it in the Readme.

**Comment by shazow** [comment]
@mining-visualizer Hiya, what version of vipnode are you using? Could you paste the first 10+ lines of the log? (Including the initialization/connecting/etc)

And yes, the instructions should still be there! Opened #61 to investigate.

**Comment by mining-visualizer** [comment]
I'm using v2.1. Running Ubuntu under Windows 10, ie. WSL.   Here's the program output:

```
2019-07-11 19:11:12.311 INFO Connecting to Ethereum node: /home/russel/.ethereum/geth.ipc
2019-07-11 19:11:12.313 INFO Connecting to remote pool: wss://pool.vipnode.org/
[agent] 2019/07/11 19:11:14 Connected to local geth-light node: enode://30bd5a214285637e900065f4308c7372d3cdf6acca8e2ae981be9b7b45885615b1724bf968c24f1addab569a8b16d6336b02a9235f20f053063d21922ab0cd0f@127.0.0.1:30303?discport=0
[agent] 2019/07/11 19:11:14 Registered on pool: Version vipnode/pool/v2.1
2019-07-11 19:11:14.777 ALERT Message from pool: $(cat /home/ubuntu/motd.txt)
[agent] 2019/07/11 19:11:15 Pool update: peers=0 active=0 invalid=0 block=0 balance=Balance(<null account>, 0 wei)
2019-07-11 19:11:15.080 WARNING Local node is behind the latest known block number by 8133586 blocks: 8133586
[agent] 2019/07/11 19:12:15 Pool update: peers=0 active=0 invalid=0 block=0 balance=Balance(<null account>, 0 wei)
```

**Comment by mining-visualizer** [comment]
I tried to find that alert message from the pool, but I don't have a folder `/home/ubuntu`.

**Comment by shazow** [comment]
@mining-visualizer Nope, that's totally on me! I messed up the deployment. I'll fix it tomorrow. :) Let's move this discussion to #61

**Comment by shazow** [comment]
This particular bug is fixed in master, will ping #62 when it's in a stable release and deployed.
